src/www/main.js.rst

```diff
@@ -5,22 +5,49 @@ Illustrate using a WebAssembly module compiled from Rust. Start by loading the
 
 .. code-block:: javascript
 
-    const wasm = require('../main.rs');
+    import wasm from '../lib.rs';
 
-Next, initialize the WebAssembly module and create JavaScript wrappers for the
-Rust functions contained in the module.
+This creates ``wasm`` as a Promise resolving to a `WebAssembly.Instance`_.
 
-The Rust source must have a ``main`` function or it does not compile. However,
-``exit(0)`` is implicit at the end of ``main`` which would cause the WebAssembly
-runtime to exit. Pass ``noExitRuntime`` to the module initialisation to suppress
-this behaviour.
+.. _WebAssembly.Instance: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Instance
+
+Unlike Emscripten, the LLVM backend for wasm-unknown-unknown target does not
+provide linking for math functions. More details are in this
+`StackOverflow question <https://stackoverflow.com/questions/47997306/rust-wasm32-unknown-unknown-math-functions-not-linking>`_.
+
+To get round this, we provide JavaScript implementations for the missing
+functions and the WebAssembly loader inserts these into the environment for the
+loaded module.
+
+The Rust random number generator causes an exception when used in WebAssembly
+modules because it relies on ``/dev/random`` for implementation. Again, we stub
```

Comment:
That's a bad idea to begin with. `urandom` or bust.

Reply:
I'll rephrase this since I don't know whether it is actually using `/dev/urandom`. Not everybody is as accurate as they could be about the difference.